api/v1alpha1/syncconfig_types.go

```diff
@@ -22,6 +14,7 @@ type (
 		// NamespaceSelector defines which namespaces should be targeted
 		NamespaceSelector *NamespaceSelector `json:"namespaceSelector,omitempty"`
 		// SyncItems lists items to be synced to targeted namespaces
+		// +kubebuilder:pruning:PreserveUnknownFields
```

Comment:
```suggestion
		// +kubebuilder:pruning:PreserveUnknownFields
		// +kubebuilder:validation:EmbeddedResource
```

Would this maybe help to get rid of the mentioned validation errors?
https://book.kubebuilder.io/reference/markers/crd-validation.html

Reply:
uuuh, lemme try

Reply:
I think it's having a problem with this property being an array:
```
The CustomResourceDefinition "syncconfigs.sync.appuio.ch" is invalid: spec.validation.openAPIV3Schema.properties[spec].properties[syncItems].type: Invalid value: "array": must be object if x-kubernetes-embedded-resource is true
```
```
		// +kubebuilder:pruning:PreserveUnknownFields
		// +kubebuilder:validation:EmbeddedResource

		// SyncItems lists items to be synced to targeted namespaces
		SyncItems []unstructured.Unstructured `json:"syncItems,omitempty"`
```

Reply:
We might be able to define a type alias for `unstructured.Unstructured` and set the validation on it?

Reply:
I think I was able to get rid of the validation error by aliasing `Unstructured` and put the preserveUnknownFields marker there.

Reply:
Can you try with `// +kubebuilder:validation:EmbeddedResource` as well? Would be nice to have some basic validation that the sync items include valid K8s objects.

Reply:
I put that marker in both `SyncItems []SyncItem` and `SyncItem`. `make generate` did not add any other fields to the CRD spec except for preserveUnknownFields, so I left it out again.
Do you think we should do it for forward compatibility?

Reply:
I left it in there now, even it doesn't do anything now.
`// +kubebuilder:validation:EmbeddedResource` may be working if `Unstructured` is used as its own field, but not in an array.